### ISSUE TITLE
DEP-1 — ASE calculator for the v1 engine (committee, Hessians, descriptors, key pass-through)

### ISSUE THREAD

**Issue by aacostadiaz** [ACEsuit/mace#1583]
**Depends on:** ARCH-4 (#1563) · **Blocks:** CLI-1 (#1579), DEP-1a (#1634), FM-3 (#1588), DEP-3 (#1595), GATE-3 (#1585)

**Prerequisite:** the Phase 0 characterization suite is committed — goldens, fixtures and the single-source tolerance table under `tests/golden/`, characterization tests under `tests/unit/`. References of the form `P0-N` name that work.

**Sequencing:** the calculator consumes the two-phase forward + derivative engine (forces via autograd, stress via strain injected before edge vectors), so it cannot begin before ARCH-4 (#1563) lands.

**Context:** The ASE calculator is the successor of `mace/calculators/mace.py` (1,551 lines, of which `MACECalculator` is `:77-1004` and the separate `MagneticMACECalculator` is `:1006-1551`): the primary Python-side way users run a trained MACE potential in MD/relaxation. The legacy class mixes concerns the v1 stack now separates — it loads backend `state_dict`s directly, dispatches on a `model_type` string across five values (`:193-244`), hand-keeps the node-level/graph-level output key sets used to slice padded results (`_slice_real_outputs`, `:459-502`), and hard-codes padding heuristics (`_EDGE_PAD_MULTIPLE = 64`, `_EDGE_PAD_HEADROOM = 1.25` at `:59-60`, auto-estimate at `:567-582`, grow-and-recompile on overflow at `:607-619`). This ticket owns the **energy family** — `model_type="MACE"` and its committee, padding, units, pass-through, Hessian and descriptor APIs. The non-energy families (dipole/dielectric/polar/LES) and the magnetic calculator are DEP-1a (#1634); splitting them keeps this ticket at Size L and makes the family surface an explicit deliverable rather than an out-of-scope line. The E2E behavioural contract is the P0-5 calculator contract (keys `energy`, `forces`, `stress` when periodic; shapes/dtypes; numeric agreement with `mace eval` on the same configs). Legacy is read in place as the behavioural oracle; reproduce its numbers and result keys, never its structure. Active-learning MD (consumed later by CLI-1 (#1579)'s `mace md active-learning`) depends on this ticket's committee mean/variance support.

**Interface & constraints:**

1. **One loader, shared with every non-LAMMPS consumer.** The calculator wraps the deploy loader rather than owning model loading:

```python
# mace_torch.deploy
def load_deployed(path: str, *, device: str = "cpu") -> "DeployedModel":
    """Returns the built eager model + its metadata; .compute(graph) -> MACEOutputs."""
```

**This ticket ships the minimal `load_deployed`,** and owns it: given a path to a neutral checkpoint (canonical safetensors weights plus the JSON sidecar carrying the model config and the resolved config), it rebuilds the eager model through `mace_torch.serialization.load_checkpoint` and returns it with that metadata. There is **no deployment manifest and no bundle directory at this point** — the LAMMPS bundle and its `deploy.json` are DEP-2 (#1594), which extends this same function to accept a bundle path rather than replacing it. Do not invent a manifest here.

Model artifacts are self-contained potentials: E0s live inside the model, so the calculator performs **no E0 add-back** and duplicates no energy algebra. (Per-atom results keep the legacy split: `results["energies"]` is the full node energy including E0, while `results["node_energy"]` has the per-atom E0 subtracted — `mace/calculators/mace.py:802-804`.)

2. **Padding is a policy whose default is `none` — matching develop, not the other way round.** Legacy pads only when asked: `pad_num_atoms`/`pad_num_edges` default to 0 (`mace/calculators/mace.py:112-113`), fall back to the env vars `MACE_ASE_PAD_NUM_ATOMS` / `MACE_ASE_PAD_NUM_EDGES` (`:422-425`), and **auto-estimation runs only under `torch.compile`** (`if self.use_compile and not self._padding_initialized`, `:605-606`). The grow-on-overflow branch likewise fires only once padding is initialized (`:608-619`). v1 keeps those three regimes exactly and makes them explicit:

```python
@dataclass(frozen=True)
class PaddingPolicy:
    mode: Literal["none", "fixed", "auto"] = "none"   # develop's default is no padding
    nodes_budget: int = 0
    edges_budget: int = 0
    graphs_budget: int = 0
    edge_multiple: int = 64      # legacy _EDGE_PAD_MULTIPLE (mace/calculators/mace.py:66)
    headroom: float = 1.25       # legacy _EDGE_PAD_HEADROOM (mace/calculators/mace.py:67)
    on_overflow: Literal["grow", "error"] = "grow"   # grow = bump budget + one recompile (:596-608)

def pad_batch(data: dict, policy: PaddingPolicy, r_max: float) -> tuple[dict, PaddingInfo]: ...
def unpad_outputs(outputs: "MACEOutputs", info: PaddingInfo) -> "MACEOutputs": ...
```

`mode="auto"` is what the compiled path selects (BKD-2 (#1566)'s `dynamic=False` requirement is the reason padding exists at all), and the auto budget reproduces the legacy heuristic: node budget = the first call's real node count, edge budget = `round_up(real_edges * headroom, edge_multiple)`. `unpad_outputs` learns which field is node-level vs graph-level from the observable spec (CORE-1 (#1555)) — the calculator must not re-grow a hand-kept key list like `_slice_real_outputs` (`:463-490`), which in develop lists 22 keys across two sets (8 graph-level, 14 atom-level) and had to be extended for LES and Fukui outputs — and still classifies only 22 of the 43 keys a model can emit, the other 21 falling through the `else` at `:500-501` with their padding rows intact. Invariant (numeric): outputs on real slots are identical with and without padding at the per-dtype tolerance — padding never reorders, it only appends. `tests/unit/test_padding.py` and `tests/workflows/test_calculator.py:591-592` are the behavioural floor.

3. **Committee/ensemble.** Accept multiple models (list, `Path`, or wildcard string, as legacy `model_paths` with `mace_*.model`-style globs, `mace/calculators/mace.py:247-261`). All committee members must have the same `r_max`; a mismatch is a hard error naming the values (`:298-302`). Result semantics (`:771-828`):
   - standard ASE keys (`energy`, `forces`, `stress`, …) hold the **mean** over models;
   - `<key>_comm` holds the per-model stack and `<key>_var` the **population variance** (`unbiased=False`), for the four keys legacy stores an ensemble of: `results_store_ensemble = {energy, forces, stress, dipole}` (`:729`). The loop emits **both** suffixes for all four (`:771`, `:781`), so a committee run writes eight keys — while `implemented_properties` advertises only `energy_comm`, `energy_var`, `forces_comm`, `stress_var` (`:278-287`, `:293`). Reproduce the eight the code emits, not the four it declares, and close the gap by declaring all eight. v1 may widen the ensemble set beyond those four keys, but it must do so deliberately and say so — silently emitting `<key>_var` for every key is a behaviour change;
   - `free_energy` aliases `energy`; stress is converted 3×3 → Voigt 6 (`full_3x3_to_voigt_6_stress`), and `stresses` (per-atom) is converted row-wise (`:805-830`).

4. **Units and pass-through.** Constructor options `energy_units_to_eV` and `length_units_to_A` (defaults 1.0, `:101-102`) convert model units to ASE units per quantity through the `results_map` table (`:729-752`): energies × E, forces × E/L, stress / `stresses` / `virials` × E/L³, and dipole / charges / polarizability × 1.0. `info_keys`/`arrays_keys` dicts map `atoms.info`/`atoms.arrays` entries into graph input fields; `charges_key` (default `"Qs"`, `:104`) is folded into `arrays_keys` at `:576`. `check_state` must treat pass-through `atoms.info` values as state so ASE's result cache invalidates when they change (legacy override at `:430-456`, which compares `atoms.info` key-by-key and skips ndarray values).

5. **Extra evaluation APIs**, ported with legacy signatures/shapes:
   - `get_hessian(atoms)` (`:929`) — returns `np.ndarray` for one model, `list` for a committee;
   - `get_descriptors(atoms, invariants_only=True, num_layers=-1)` → `np.ndarray` of shape `(num_atoms, num_interactions, invariant_features)` for a single model, `list[np.ndarray]` for a committee (`:953-1006`).
   Both are `model_type`-gated in legacy; in v1 the gate is the model's declared observables, and the "not implemented for this family" error must still be a clear refusal (the DEP-1a (#1634) families extend the gate).

6. **Layering:** `packages/` never imports the legacy `mace/` (allowlist: `mace_launcher`, `tests/parity/`); legacy-vs-v1 numeric comparisons live in `tests/parity/` only.

**Task:**
1. Implement `mace_torch/calculators/ase_calculator.py`: ASE `Calculator` subclass wrapping `mace_torch.deploy.load_deployed`; graph construction from `ase.Atoms` through the v1 graph contract and the configured neighbor backend; `check_state` including the info-dict comparison.
2. Wire `PaddingPolicy` with all three legacy regimes (`none` default, `fixed` from explicit budgets or the two env vars, `auto` under compile with headroom 1.25 / multiple 64 and grow-on-overflow at most one recompile per growth) and `unpad_outputs` driven by the observable spec.
3. Committee support: multi-model load (list, `Path`, wildcard), same-`r_max` validation, mean into standard keys, `<key>_comm` / `<key>_var` (population variance) for the pinned ensemble key set.
4. Unit-conversion table (E, E/L, E/L³, and the unity entries), `free_energy` alias, Voigt stress for both `stress` and `stresses`, the `energies`/`node_energy` per-atom split, and `info_keys`/`arrays_keys`/`charges_key` pass-through.
5. Port `get_hessian` and `get_descriptors` with legacy shapes and committee behaviour, gated on declared observables rather than a `model_type` string.
6. Tests: unit tests under `packages/mace-torch/tests/calculators/`; a parity case in `tests/parity/` comparing energy/forces/stress and `get_descriptors`/`get_hessian` against the legacy calculator on the tiny anchors at fp64; re-run the P0-5 ASE-calculator contract under the v1 engine.

**Out of scope:** the non-energy model families and the magnetic path (DEP-1a (#1634)); LAMMPS/MLIAP runtime adapters and export (DEP-2 (#1594)); OpenMM and other secondary integrations (DEP-3 (#1595)); foundation-model loaders `mace_mp`/`mace_off`/`mace_mdp` equivalents (FM-3 (#1588)); the `mace md active-learning` CLI (CLI-1 (#1579)).

**Acceptance criteria:**
- [ ] The P0-5 ASE-calculator contract passes against the v1 calculator: keys `energy`, `forces`, `stress` (when periodic) with correct shapes/dtypes, and numeric agreement with `mace eval` on the same configs.
- [ ] v1 calculator agrees with the legacy calculator on the tiny anchors at fp64 tolerance (energy, forces, stress, descriptors, Hessian) via `tests/parity/`.
- [ ] Default construction pads nothing; explicit budgets and the two `MACE_ASE_PAD_*` env vars select `fixed`; the compiled path selects `auto`. Padded and unpadded evaluations produce identical outputs on real slots at the per-dtype tolerance, and an edge-budget overflow grows the budget and triggers at most one recompile.
- [ ] Committee of N models: standard keys are the mean; `<key>_comm` **and** `<key>_var` appear for all four members of the pinned ensemble key set (or a deliberately widened set recorded in the PR), with `unbiased=False`, and every emitted key is declared in `implemented_properties`; mixed `r_max` across members raises a hard error naming the values.
- [ ] The calculator's `results` keys are enumerated against CORE-1 (#1555) constraint 7 layer (b): every energy-family name the legacy calculator can write is produced or carries a recorded DROP — including the calculator-only names `free_energy`, `energies`, `stresses` and the eight committee keys, none of which exist in any model forward.
- [ ] Unit conversions applied per quantity (E, E/L, E/L³, unity); `free_energy` mirrors `energy`; `stress` and `stresses` returned in Voigt 6; `energies` (with E0) vs `node_energy` (E0-subtracted) split preserved.
- [ ] `info_keys`/`arrays_keys`/`charges_key` values reach the graph inputs, and changing a passed-through `atoms.info` value invalidates the ASE result cache.
- [ ] The calculator contains no hand-kept node-vs-graph key list and no E0 arithmetic; `packages/` does not import legacy `mace/`.

**Inventory gaps assigned here:**
- [ ] Committee support in `calculate` (multiple models → mean/variance) — the part of `mace_active_learning_md` that survives its DROP, and the reason the MD driver could be dropped at all. `tests/workflows/test_calculator.py` exercises a committee but not the `_var` semantics.
- [ ] `energy_units_to_eV` / `length_units_to_A` — neither appears anywhere in `tests/`; the whole unit-conversion table is unpinned.
- [ ] `MACE_ASE_PAD_NUM_ATOMS` / `MACE_ASE_PAD_NUM_EDGES` — the env-var route into `fixed` padding has no test (only the constructor arguments do).

**Verify:**
```bash
python -m pytest packages/mace-torch/tests/calculators/ -v
python -m pytest tests/unit/test_padding.py tests/unit/test_calculator_dtype_scope.py -v
python -m pytest tests/parity -m "not slow" -k calculator          # legacy-vs-v1 on the anchors
MACE_ENGINE=v1 python -m pytest tests/workflows/test_calculator.py -v   # P0-5 contract
```
**Review focus:** numeric agreement with the legacy calculator (fp64 parity on the anchors), the padding default (develop pads nothing unless asked — an "auto by default" v1 would change every uncompiled user's graph sizes), and the committee variance semantics: a silent mean/variance or unit-conversion slip corrupts active-learning uncertainty downstream. Also that unpadding is spec-driven, not key-list-driven.

---
